src/libstd/fs.rs

```diff
@@ -663,15 +623,17 @@ impl Permissions {
     }
 }
 
-#[unstable(feature = "file_type", reason = "recently added API")]
 impl FileType {
```

Comment:
We're currently not necessarily considering a hidden/unstable variant to be an "unstable enum", it was mostly just how `ErrorKind` worked out. The set of file types also varies quite a bit across platforms, and the extra types made somewhat more sense to provide through extension traits instead of via this `enum`.

Finally, using an opaque structure allows one day accessing the raw bits used to determine the file type which may contain some unknown information.